backend/src/nodes/nodes/image_channel/palette_extraction.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from enum import Enum
+
+import numpy as np
+
+from . import category as ImageChannelCategory
+from ...impl.dithering.palette import (
+    distinct_colors_palette,
+    kmeans_palette,
+    median_cut_palette,
+)
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties import expression
+from ...properties.inputs import ImageInput, EnumInput, NumberInput
+from ...properties.outputs import ImageOutput
+
+
+class PaletteExtractionMethod(Enum):
+    ALL = "all"
+    KMEANS = "k-means"
+    MEDIAN_CUT = "median"
+
+
+PALETTE_EXTRACTION_METHOD_LABELS = {
+    PaletteExtractionMethod.ALL: "All distinct colors",
+    PaletteExtractionMethod.KMEANS: "K-Means",
+    PaletteExtractionMethod.MEDIAN_CUT: "Median cut",
+}
+
+
+@NodeFactory.register("chainner:image:palette_from_image")
+class PaletteFromImage(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = (
+            "Use an image to create a palette.  A palette is an image with one row."
+        )
+        self.inputs = [
+            ImageInput(),
+            EnumInput(
+                PaletteExtractionMethod,
+                option_labels=PALETTE_EXTRACTION_METHOD_LABELS,
+                default_value=PaletteExtractionMethod.KMEANS,
+            ).with_id(1),
+            group(
+                "conditional-enum",
+                {
+                    "enum": 1,
+                    "conditions": [
+                        [
+                            PaletteExtractionMethod.KMEANS.value,
+                            PaletteExtractionMethod.MEDIAN_CUT.value,
+                        ]
+                    ],
+                },
+            )(
+                NumberInput(
+                    "Palette Size",
+                    minimum=2,
+                    default=8,
+                ).with_id(2),
+            ),
+        ]
+        self.outputs = [
+            ImageOutput(
+                image_type=expression.Image(
+                    width="""
+                match Input1 {
+                    PaletteExtractionMethod::All => int(1..),
+                    _ => Input2
```

Comment:
The palette size fix you recently pushed invalidates this type. If the images has, let's say, 3 distinct colors and the user set a palette size of 4, then your fix will return a 3x1px image, right?

Reply:
It should still be 4x1, since I'm padding it:

https://github.com/chaiNNer-org/chaiNNer/blob/eb457466cc68a7d6777ba9a10b3d976ca452ae54/backend/src/nodes/nodes/image_channel/palette_extraction.py#L97-L99

Reply:
This test suggests it's working as intended:  

extracting 10 colors from a 8x1 image:  ![image](https://user-images.githubusercontent.com/551173/216420577-dfdd9ff1-a109-4050-bda1-189dbc6e2faf.png)

10x1 output:  ![image](https://user-images.githubusercontent.com/551173/216420750-827fc8f8-ef18-446c-aaf6-9d1a6f85f0a9.png)

Reply:
Sorry to sneak that in after it's already reviewed.  I'll open another PR next time.

Reply:
> It should still be 4x1, since I'm padding it

Ah, I didn't see that part. My bad.

> Sorry to sneak that in after it's already reviewed. I'll open another PR next time.

No, don't worry about it. GH automatically dismisses my review when you push new commits, so I would have reviewed them anyway.